<launch>

 <arg name="fiducial_len" default="0.062"/>

 <!-- Setup to publish our robot model -->
 <arg name="model" default="droidBot.urdf" />

 <node pkg="tf" type="static_transform_publisher" name="map_to_odom" args="0.0 0.0 0.0 0 0 0.0 /map /odom 20"/>
 <node pkg="tf" type="static_transform_publisher" name="base_frame_laser" args="-0.03 0 0.20 0 0 0 /base_link /laser 20"/>
 <node pkg="tf" type="static_transform_publisher" name="laser_prox_sensors" args="0.0 0 0.10 0 0 0 /laser /sensor_axis 20"/>

 <param name="robot_description" textfile="/home/ubuntu/catkin_ws/droidBot.urdf" />

 <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
   <param name="publish_frequency" value="20.0"/>
 </node>

 <!--  Ubiquity raspicam_node                 -->
 <node type="raspicam_node" pkg="raspicam_node" name="raspicam_node" output="screen">
    <!--   I had a camera yaml file like this before cameraRaspicam1p3_1280x720.yaml -->
    <param name="camera_info_url" value="package://raspicam_node/camera_info/camerav2_1280x720.yaml"/>
    <param name="width" value="1280"/>
    <param name="height" value="720"/>
    <!-- Low frame rate less cpu load but more motion blur -->
    <param name="framerate" value="4"/>
    <param name="exposure_mode" value="antishake"/>
    <param name="shutter_speed" value="0"/>
    <param name="camera_frame_id" value="raspicam"/>
 </node>

 <!--  Mark-Toys.com Motor controller  TofRadar, Edge detect    -->
 <node pkg="robo_mag" type="base_control" name="base_control" output="screen" respawn="false">
       <param name="/edge_detect_type" value="1" />
       <param name="/tof_radar_slots" value="11" />
       <param name="/tof_radar_fov_width_rads" value="0.25" />
       <param name="/tof_radar_spacing_rads" value="0.19" />
       <param name="/serial_port" value="/dev/ttyUSB0"/>
       <param name="/baudrate"    value="115200"/>
       <param name="/pid_enabled" value="0" />
       <param name="/pid_Kp"      value="7000.0" />
       <param name="/pid_Ki"      value="1500.0" />
       <param name="/pid_Kd"      value="0.0" />
       <param name="/pid_intCap"  value="400.0" />
       <param name="/pid_Kdom"    value="11000.0" />
       <param name="/enc_tic_per_rev"  value="8600.0" />
       <param name="/wheel_spacing"    value="0.252" />
       <param name="/wheel_diameter"   value="0.072" />
 </node>

  <!--  Start up the Robo Magellian Nodes                    -->
 <node pkg="robo_mag" type="robo_mag_main" name="robo_mag_main" output="screen" respawn="true">

        <!--  Robot Type defines the hardware.                  -->
        <!--  1=RoboMadgeEllen  2=DroidBot                      -->
        <param name="/robot_mode"            value="2" />

        <!--  Robo Magellian Mode. Controls high level mode     -->
        <!--  0 - General purpose bot acting on instructions    -->
        <!--  1 - Robo Magellan mode is a future mode for GPS   -->
        <!--  2 - Lidar Follow mode follows a nearby object     -->
        <!--  3 - Lidar Room Roam wanders around a closed room  -->
        <!--  4 - RoboMadge moves back and forth between tags   -->
        <!--  5 - DroidBot Wanders on Table                     -->
        <!--  6 - DroidBot goes back and forth (full reverse)   -->
        <!--  7 - DroidBot Finds and pushes can off of table    -->
        <!--  8 - DroidBot Finds and pushes can into goal       -->
       <param name="/robo_madge_mode"        value="0" />

        <!--  Wheel geometry related values for this bot       -->
       <param name="/bot_wheel_spacing"      value="0.23" />
       <param name="/bot_wheel_vel_scaling"  value="50.0" />
       <param name="/bot_wheel_ang_scaling"  value="6.28" />

        <!--  These are used in Lidar Follow mode              -->
        <!--  cam_follow  - Webcam follows the nearby target   -->
        <!--  bot_rotate  - Bot rotates towards closest target -->
        <!--  bot_follow  - move to closest target IF rotated  -->
       <param name="/lidar_obj_cam_follow"   value="0" />
       <param name="/lidar_obj_bot_rotate"   value="0" />
       <param name="/lidar_obj_bot_follow"   value="0" />

       <param name="/robo_nav_rotate_speed"       value="1.5" />
       <param name="/robo_robo_nav_rotate_goal"   value="10" />
       <param name="/robo_nav_distance_goal"      value="0.5" />
       <param name="/robo_nav_drive_speed_cm_sec" value="12.0" />
       <param name="/site_rough_longitude"    value="122.1" />
       <param name="/site_rough_lattitude"    value="37.25" />
       <param name="/drive_motor_idle"        value="127.0" />
       <param name="/drive_motor_range"       value="40.0" />
       <param name="/cmd_vel_use_overrides"   value="0" />
       <param name="/cmd_vel_set_linearX"     value="0.0" />
       <param name="/cmd_vel_set_angularZ"    value="0.0" />
       <param name="/steering_center"         value="108.0" />
       <param name="/steering_range"          value="140.0" />
       <param name="/cam_pan_view_center"     value="110" />
       <param name="/cam_tilt_view_center"    value="100" />
       <param name="/cam_pan_look_rate"       value="200" />
       <param name="/cam_pan_look_range"      value="0" />
       <param name="/cam_pan_follow_cone"     value="1" />
       <param name="/lidar_obj_min_range"     value="0.1" />
       <param name="/lidar_obj_max_range"     value="1.5" />
       <param name="/lidar_obj_min_deg"       value="2" />
       <param name="/lidar_obj_max_deg"       value="50" />
       <param name="/lidar_open_gap_range"    value="0.8" />
       <param name="/lidar_open_gap_width"    value="0.3" />
       <param name="/lidar_obj_rotate_speed"  value="0.7" />
       <param name="/lidar_obj_distance_goal" value="0.3" />
       <param name="/lidar_obj_drive_speed_cm_sec"   value="9.0" />
 </node>

  <!--  XV-11 Neato Lidar Node from getsureal controller -->
  <node pkg="xv_11_laser_driver" type="neato_laser_publisher" name="neato_laser_publisher"  respawn="true">
     <param name="port" value="/dev/ttyACM0"/>
     <param name="firmware_version" value="2"/>
     <param name="frame_id" value="robo_world"/>
  </node>


</launch>

